engine/src/gnuhash.h

```diff
@@ -66,16 +67,19 @@ struct hash<std::pair<Unit *, Unit *> > {
                 ^ ((size_t) a((int) (((size_t) key.second) >> 4))));
     }
 };
-#ifdef __GNUC__
+
+}
```

Comment:
NOTE: We really should derive a new type from `std::hash` and add the above to it instead of extending `std::hash` to do what we want. Not sure if that's viable in this PR or not, but something to think about.

Reply:
@BenjamenMeyer I agree with the concept. I'm trying to make one small change at a time though, to minimize breakage. :-)